File: tatala-java/src/com/qileyuan/tatala/executor/ServerExecutor.java
package com.qileyuan.tatala.executor;

import com.qileyuan.tatala.socket.client.SocketController;
import com.qileyuan.tatala.socket.to.TransferObject;

/**
 * This class is the distribution-method-call executor, which dispatches request to socket controller.
 * 
 * @author JimT
 *
 */
public class ServerExecutor{
	public static Object execute(TransferObject to){
		return SocketController.execute(to);
	}
}

File: history/0.2.0/tatala-client-csharp/src/QiLeYuan/Tatala/executor/ServerExecutor.cs
using System;
using QiLeYuan.Tatala.util;
using QiLeYuan.Tatala.socket.to;

/**
 * This class is the distribution-method-call executor, which dispatches request to different target.
 * For now, it only supports local and socket distribution.
 * 
 * @author JimT
 *
 */
namespace QiLeYuan.Tatala.executor {

    public class ServerExecutor {
        private static ServerTarget target;
        public static Object execute(TransferObject to) {
            Object retObj = null;
            //for csharp client, server target is Socket.
            target = new SocketServerTarget();
            retObj = target.execute(to);
            return retObj;
        }
    }
}